lib/account/account-service.js

```diff
@@ -1,99 +1,128 @@
+     * when the account file is created.
      *
      * TODO: Encrypt data
      *
-     * @param {string} accoundId
      * @param {Object} accountInfo
      * @param {Object} options
-     * @return {Promise.<Object>}
+     * @param {string} regId
+     * @return {Promise.<accountManager~completionInfo>}
      */
-    completeAsync(directory, accountInfo, options) {
-        return this.getAsync(directory).then((accountFile) => {
-            accountFile.password = accountInfo.password;
+    function completeAsync(accountInfo, options, regId) {
+        return getDirectoryAsync(registrationDirName, regId).then((regDirectory) => {
+            return getAsync(regDirectory).then((accountFile) => {
+                accountFile.accountId = accountInfo.accountId;
+                accountFile.password = accountInfo.password;
+                delete accountFile.regId;
 
-            return this.storage.putAsync(directory, JSON.stringify(this.accountFile), options).then(() => {
-                return {
-                    accountId: accountFile.accountId
-                };
+                return makeAccountFileAsync(accountFile, options).then(() => {
+                    return storage.delAsync(regDirectory).then(() => {
+                        return {
+                            accountId: accountInfo.accountId
+                        };
+                    });
+                });
             });
         });
     }
 
 
     /**
+     * Creates the account file in the account directory.
+     *
+     * @param {Object} accountInfo
+     * @param {Object} options
+     * @return {Promise.<boolean>}
+     */
+    function makeAccountFileAsync(accountInfo, options) {
+        return getDirectoryAsync(accountDirName, accountInfo.accountId).then((accountDir) => {
+            return storage.putAsync(accountDir, JSON.stringify(accountInfo), options);
+        });
+    }
+
+
+    /**
      * Gets the account file from the directory.
      *
      * TODO: Decrypt data
      *
-     * @param {string} accountId
+     * @param {string} directory
      * @return {Promise.<AccountService~accountFile>}
      */
-    getAsync(directory) {
-        return this.storage.getAsync(directory).then((accountFile) => {
+    function getAsync(directory) {
+        return storage.getAsync(directory).then((accountFile) => {
             return JSON.parse(accountFile.toString("binary"));
         });
     }
 
 
     /**
+     * Get the registration file when it's still in the registration directory.
+     * This is to eliminate one steps for the manager getting the directory
+     * and then requesting the file.
+     *
+     * @param {string} regId
+     * @return {Promise.<accoutService~accountFile>}
+     */
+    function getRegistrationFileAsync(regId) {
+        return getDirectoryAsync(registrationDirName, regId).then((directory) => {
+            return getAsync(directory).then((accountFile) => {
```

Comment:
Same here as above.

```
return getDirectoryAsync(registrationDirName, regId).then(getAsync);
```

Also the one callback that just returns the value that was passed in is unneeded.